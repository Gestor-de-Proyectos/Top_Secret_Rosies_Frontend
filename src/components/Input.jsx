import React from 'react';

const Input = ({ label, name, defaultValue, type, required }) => {
  return (
    <label htmlFor={name} className='flex flex-col my-3'>
      <span>{label}</span>
      <input
        required={required}
        type={type}
        name={name}
        className='input'
        defaultValue={defaultValue}
        data-testid='input-test'
      />
    </label>
  );
};

export default Input;
