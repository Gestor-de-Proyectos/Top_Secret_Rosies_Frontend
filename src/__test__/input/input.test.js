import React from 'react';
import Input from 'components/Input';


import { render, cleanup, screen } from '@testing-library/react';


afterEach(cleanup);
it('renders okay', () => {
  render(<Input loading={false} text={'Prueba'} />);
  expect(screen.getByTestId('input-test')).toBeInTheDocument();
});


it('renders svg when loading is activated', () => {
  render(<Input disabled={false} loading={true} text={'Hola'} />);
  expect(screen.getByTestId('input-test')).toMatchSnapshot();
});