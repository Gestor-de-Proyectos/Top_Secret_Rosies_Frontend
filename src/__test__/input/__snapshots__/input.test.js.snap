// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders svg when loading is activated 1`] = `
<input
  class="input"
  data-testid="input-test"
  value=""
/>
`;
